[AIRFLOW-510] Hide "paused" DAGs on the HomeView unless "Shown"

btallman opened the pull request:
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- _https://issues.apache.org/jira/browse/AIRFLOW-510_

Testing Done:
- UI only, and as far as I can tell, we don't have UI test coverage yet

![screen shot 2016-09-14 at 10 55 23 pm](https://cloud.githubusercontent.com/assets/173950/18539745/7d3c105a-7ace-11e6-9aa1-26d6258d1aaf.png)
![screen shot 2016-09-14 at 10 55 36 pm](https://cloud.githubusercontent.com/assets/173950/18539749/808a6fa4-7ace-11e6-9969-a6c5ed0d52f6.png)

ldct commented on airflow/www/templates/airflow/dags.html:
```diff
@@ -184,10 +189,11 @@
         });
       });
-      $('#dags').dataTable({
+      var dt = $('#dags').dataTable({
```
Use a more descriptive variable name

ldct commented on airflow/www/templates/airflow/dags.html:
```diff
@@ -184,10 +189,11 @@
         "iDisplayLength": 500,
         "bSort": false,
       });
+//      debugger;
```
what's this?

ldct commented on airflow/www/views.py:
```diff
@@ -1818,7 +1823,12 @@ def index(self):
-        unfiltered_webserver_dags = [dag for dag in dagbag.dags.values() if not dag.parent_dag]
+        # optionally filter out "paused" dags
+        if request.args.get('showPaused') == 'True':
+            unfiltered_webserver_dags = [dag for dag in dagbag.dags.values() if not dag.parent_dag]
```
if we're adding yet another filter to the existing ones we need to rename the variables `do_filter`, `unfiltered_webserver_dags` etc

btallman committed:
Add Support for dag.backfill=(True|False) Option

Added a dag.backfill option and modified the scheduler to look at
the value when scheduling DagRuns (by moving dag.start_date up to
dag.previous_schedule), and added a config option scheduler.backfill_by_default (defaults to True)
that allows users to set this to False for all dags modifying the existing DAGs

In addition, we added a test to jobs.py (test_dag_backfill_option)
